### ISSUE TITLE
Is there any logic about improve model by self defined labels or update record fine tuning ?

### ISSUE THREAD

**Issue by svjack** [argilla-io/argilla#239]
I only find some logic about load and visualize.
And the logic about snorkel usage is simply add some label functions by 
observe without update pretrain models.
Is  there any logic about improving user defined model ?

streaming seems only for edit.

**Comment by dvsrepo** [comment]
Hi @svjack , thanks for asking!

We are working on a guide for using annotations to improve/fine-tune an existing model. It's true this is missing.

For now a few pointers on how this workflow could look like:

1. In the active learning tutorial, here's the logic for (1) reading a rubrix dataset (containing annotations), and (2) creating a training set with these annotations, and (2) teaching the model with the new annotations: https://docs.rubrix.ml/en/stable/tutorials/05-active_learning.html#2.-Teach-the-learner

2. In the Cookbook, here's the logic for (1) reading a rubrix dataset containing labels, (2) creating a hugging face dataset, and (3) fine-tuning a pre-trained hugging face model: https://docs.rubrix.ml/en/stable/guides/cookbook.html#Training

Annotations can be added in two ways:

1. Programmatically via `rb.log()` where you can use weak supervision, rules, an existing training dataset.

2. Manually with the UI annotation mode, where you can use "bulk" labeling and search-features (https://docs.rubrix.ml/en/stable/reference/rubrix_webapp_reference.html#annotation-mode)

Every time you do `rb.load()` you'll get the current annotations and you can fine-tune/train your model. Then, you can build workflows for using your fine-tuned model on more data, label again and fine-tune.

Hope this is useful for you, I'll ping you when we have the more detailed guide.

**Comment by dvsrepo** [comment]
Hi @svjack , as promised, here's a tutorial for fine-tuning your models using labels from Rubrix datasets:

https://docs.rubrix.ml/en/master/tutorials/06-labeling-finetuning.html

If you are not using hugging face transformers, I'd be more than happy to guide you with your use case and library.

It will go through another round of reviews, but wanted to share it with you as soon as possible.

**Comment by dvsrepo** [comment]
Hi @svjack did you get a chance to take a look at the tutorial, please do not hesitate about raising more questions.

**Comment by dvsrepo** [comment]
@svjack I'm closing this, but feel free to open new issues or threads in the Discussion forum.
